### PR TITLE
Reflect changes in cfgrammar.

### PR DESCRIPTION
**Pull request by ltratt**
This PR reflects the move of some code to lrtable, and the consequent API changes.

### CODE REVIEW

**Comment by ltratt** [comment]
[I've rekicked the build, which hopefully should now work after the cfgrammar change was merged.]

File: src/lib/firsts.rs [hunk 1]
```diff
@@ -37,7 +37,7 @@ use cfgrammar::{Grammar, NTIdx, Symbol, TIdx};
 use cfgrammar::yacc::YaccGrammar;
 
 // Firsts stores all the first sets for a given grammar. For example, given this code and grammar:
```
Comment (snim2) [inline src/lib/firsts.rs]: When I read this through I thought "Firsts" should be "First". In this comment I think your first line should start: "`Firsts` stores all the first sets..." to make it clear that this is a name from the code. The block of code underneath that shows how to create a grammar needs to be surrounded by triple-quotes.

File: src/lib/pager.rs [hunk 2]
```diff
@@ -408,13 +407,13 @@ mod test {
     //     a
     //     aSb
```
Comment (snim2) [inline src/lib/pager.rs]: Similarly, this code in the comments needs triple-quotes.

**Comment by snim2** [comment]
OK, a couple of very small issues here, otherwise looks fine.

File: src/lib/firsts.rs (hunk 1 above)
Comment (ltratt) [inline src/lib/firsts.rs]: I agree with you that the comments was a bit confusing, so I've tried fixing it.

Triple quotes aren't needed for multiline strings in Rust though.

**Comment by ltratt** [comment]
If that tweak works for you -- and if you agree about the triple quoting -- then I'll need to squash this before any merge.

**Comment by snim2** [comment]
One of your comments got lost in a folded "outdated comments" block, so I'll reply here...

I can't quite tell whether we disagree about these comments or maybe I just didn't explain very well. This is what I was expecting in the last commit as a response to the first comment:

```rust
 // This struct stores all the first sets for a given grammar. For example, given this code and
 // grammar:
 //
 //   ```
 //  let grm = YaccGrammar::new(yacc_grm(YaccKind::Original, "
 //     S: A 'b';
 //     A: 'a'
 //      | ;"));
 //   let firsts = Firsts::new(&grm);
 // ``` 
 //
 // then the following assertions (and only the following assertions) about the firsts set are
 // correct:
 //
 //   ```
 //   assert!(firsts.is_set(grm.nonterminal_off("S"), grm.terminal_off("a")));
 //   assert!(firsts.is_set(grm.nonterminal_off("S"), grm.terminal_off("b")));
 //   assert!(firsts.is_set(grm.nonterminal_off("A"), grm.terminal_off("a")));
 //   assert!(firsts.is_epsilon_set(grm.nonterminal_off("A")));
 //  ```
```

Having said that I now notice that this isn't a doc-comment, but since it is a comment against a public struct, should it be?

**Comment by ltratt** [comment]
Ah, you're right, I misunderstood. But since these are internal comments, and not doc comments, I'm not sure that markdown makes sense.

**Comment by snim2** [comment]
Should they be doc comments if they are against public structs though?

**Comment by ltratt** [comment]
In general, no; but you're right, in this case it wouldn't hurt. I'll make the change.

**Comment by snim2** [comment]
Time to squash, I think!

**Comment by ltratt** [comment]
Since the second commit isn't really related to the first, I suggest we merge with these two commits.

**Pull request merged**